features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/source/src/app/components/Applications/Details/ProductionKeys.js

```diff
@@ -34,25 +39,55 @@ class ProductionKeys extends React.Component {
         this.state = {
             application: null,
             showCS: false, // Show Consumer Secret flag
-            showAT: false,// Show Access Token flag
+            showAT: false// Show Access Token flag
         };
+        this.key_type = this.props.location.pathname.split('/').pop() == "productionkeys" 
+            ? Application.KEY_TYPES.PRODUCTION : Application.KEY_TYPES.SANDBOX;
         this.appId = this.props.match.params.applicationId;
         this.handleShowToken = this.handleShowToken.bind(this);
+        this.handleCheckboxChange = this.handleCheckboxChange.bind(this);
+        this.handleTextChange = this.handleTextChange.bind(this);
+        this.handleUpdateToken = this.handleUpdateToken.bind(this);
     }
 
     handleClickToken() {
         let application = this.state.application;
-        const type = Application.KEY_TYPES.PRODUCTION;
-        application.generateKeys(type).then(
-            () => application.generateToken(type).then(() => this.setState({}))
+        const keys = application.keys.get(this.key_type) ||
+            {
+                "supportedGrantTypes":
+                    ["client_credentials"]
+            }
+        if (keys.callbackUrl == null){
+            keys.callbackUrl = "https://wso2.am.com";
+        }
+        application.generateKeys(this.key_type, keys.supportedGrantTypes, keys.callbackUrl).then(
+            () => application.generateToken(this.key_type).then(() => this.setState({ application: application }))
         ).catch(
             error => {
                 if (process.env.NODE_ENV !== "production") {
                     console.log(error);
                 }
                 let status = error.status;
                 if (status === 404) {
-                    this.setState({notFound: true});
+                    this.setState({ notFound: true });
+                }
+            }
+        );
+    }
+
+    handleUpdateToken() {
+        let application = this.state.application;
```

Comment:
can't we use `const` instead of `let` here? I think we should change all such cases.